### ISSUE TITLE
Should `fix_python_shebang_for="bin/*"` be default in `PythonPackage`?

### ISSUE THREAD

**Issue by VRehnberg** [easybuilders/easybuild-easyblocks#3498]
Pip installed python scripts will (as far as I can [tell](https://github.com/pypa/distutils/blob/main/distutils/command/build_scripts.py#L111-L117)) replace the shebang with a hardcoded path to the Python used in the installation.

This means that `virtualenv --system-site-packages` won't be picked up.

The easy fix for these are to add:
```
fix_python_shebang_for = ['bin/*']
```
in the easyconfig.

But, looking at what we had installed on our cluster this is *not* done for a lot of softwares.

I suggest that either:
 - add this as default
 - have a sanity check to catch when this hasn't been added but should've

**Comment by boegel** [comment]
I think it makes sense to use this as default, especially since we're quite careful to not make change when there's no need for it (see also https://github.com/easybuilders/easybuild-framework/pull/4209 + https://github.com/easybuilders/easybuild-framework/pull/3208).

**Comment by VRehnberg** [comment]
Created a PR https://github.com/easybuilders/easybuild-easyblocks/pull/3499

**Comment by boegel** [comment]
fixed with #3499

**Comment by backelj** [comment]
Hi all, I encountered an issue trying to install an easyconfig (for the record: ReFrame) on our Rocky Linux 9 node.
I found the culprit to be in this adaption: the default behaviour nowadays is to replace the shebang line `#!/usr/bin/python3` with `#!/usr/bin/env python`, but since we don't have the `python-unversioned-command` package installed, and since we also don't have alternatives set, this fails because no executable named python exists...

Therefore, shouldn't this change at least include a test as well, to make sure that the replacement will indeed work? Or, alternatively, make sure it does not drop the number from the executable, i..e. let it replace `#!/usr/bin/python3` with `#1/usr/bin/env python3` instead.

**Comment by VRehnberg** [comment]
@backelj

This probably deserves its own issue (you can reference this issue from that one). Here are some quick answers (though, I'm not an EasyBuild maintainer)

> shouldn't this change at least include a test as well,

Tests for this PR where run in #3499 (though, I might have missed testing an easyconfig using system Python). Testing that the reframe executable runs is done in `sanity_check_commands` for that easyconfig. should be done in the sanitycheck for that easyconfig.

EasyConfigs for ReFrame with system Python does sanity check command `reframe -V`, if it doesn't that is an issue.

> alternatively, make sure it does not drop the number from the executable, i..e. let it replace #!/usr/bin/python3 with #1/usr/bin/env python3

That would be an issue with the behaviour of `fix_python_shebang_for` (which I'm largely ignorant about). Is probably an issue with all easyconfigs that use system python.

So in summary the possible issues I see are:

1. ReFrames sanity_check_commands doesn't fail as it should -> mostly unrelated
2. `fix_python_shebang_for` should preserve exact python version -> mostly unrelated
3. The default introduced in this PR should not be applied if system python is used  -> highly related

---

@boegel should this default only apply if system Python is not used?
